### PR TITLE
Documentation formatting / compilation fixes

### PR DESCRIPTION
**Pull request by stefanv**
Seems to work with Sphinx 2.0

### CODE REVIEW

**Comment by jni** [comment]
@stefanv does this fix the current rendering issues with arg names and types being smushed together?

**Comment by stefanv** [comment]
I didn't change anything, but it renders correctly for me.

**Comment by stefanv** [comment]
I can't replicate the Travis-CI failure on my system; anyone else?

**Comment by jni** [comment]
@stefanv I think you can download a docker image of the travis machine? That's probably the way to go. (I might be able to give that a go in a couple of weeks, but not till then.)

**Comment by hmaarrfk** [comment]
I get a similar error.
```
pickling environment... /home/mark2/git/scikit-image/doc/source/auto_examples/color_exposure/plot_rgb_to_hsv.rst:26: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/mark2/git/scikit-image/doc/source/auto_examples/segmentation/plot_thresholding.rst:21: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```
but the warning isn't being treated as an error.

**Comment by hmaarrfk** [comment]
The first comment in the file cannot be immediately after the file docstring...

File: requirements/docs.txt [hunk 1]
```diff
@@ -1,7 +1,5 @@
-# Sphinx 1.7.8 has a bug when building on cached envs
-# https://github.com/sphinx-doc/sphinx/issues/5361
-sphinx>=1.3,!=1.7.8
-numpydoc>=0.6
+sphinx>=2.0
+numpydoc==0.7
```
Comment (hmaarrfk) [inline requirements/docs.txt]: Do we know if these changes are still necessary

Comment (jni) [reply requirements/docs.txt]: We don't, but I don't think it's a big deal — docs are mostly for dev work.

Comment (jni) [reply requirements/docs.txt]: Happy to approve a different PR with just your last commit, if you want to issue such a thing. =)

Comment (hmaarrfk) [reply requirements/docs.txt]: I'll undo this if it fails. It probably will. I'm mostly worried about the pin on numpydocs

**Comment by stefanv** [comment]
There's absolutely no reason to run on an older version of numpydoc!

**Comment by stefanv** [comment]
Also, should this be considered a bug in sphinx gallery?

**Comment by jni** [comment]
@stefanv no, but no reason to force devs to update their existing environment if it's not needed, either. The requirements have a `>=` so it's fine to keep the older version supported if nothing is breaking. I'm going to go ahead and merge, we can move up any time we deem necessary. 

Re sphinx-gallery, might as well report it, but it seems like a design choice. But the error message could have been better!

**Comment by hmaarrfk** [comment]
Yeah, my greater fear is the "pin", setting `==0.7` would have made it more difficult to detect changes in numpydoc where we would need to adapt.

**Comment by stefanv** [comment]
`= 0.7` would be fine; but numpydoc gets fairly crucial fixes.  A lot of reference generation bugs have been fixed between 0.6 and 0.7, for example.

**Comment by stefanv** [comment]
@hmaarrfk I'm a bit confused; it doesn't look like my patch was proposing pinning to a specific version.  So if you're OK with `>= 0.7`, let's do that.

**Comment by hmaarrfk** [comment]
Yeah. Greater or equal is better.

I think you had `==` maybe by accident

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.15.x

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.14.x

**Pull request merged**